src/Processors/Formats/Impl/ConstantExpressionTemplate.cpp

```diff
@@ -597,6 +597,10 @@ bool ConstantExpressionTemplate::parseLiteralAndAssertType(
             if (negative || *istr.position() == '+')
                 ++istr.position();
 
+            /// Consume leading zeroes - we don't want any funny octal business
```

Comment:
The loop by bytes is slow. It has to be optimized.

Reply:
The slowness could be in the case when we speculatively try to parse a number, but it is a string starting with a lot of zeros.